src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -596,10 +610,16 @@
       },
       code: function(X) {
         var objToAdd = X.objectSummaryView ? X.objectSummaryView : X.summaryView;
+        var onExecuteFunc;
+
+        if ( X.controllerMode && X.controllerMode.equals(foam.u2.ControllerMode.SUMMARY) )
```

Comment:
For Enums, you can use ==, instead of equals(). It's faster and shorter. This works because all enum values are singletons.

Reply:
fixed